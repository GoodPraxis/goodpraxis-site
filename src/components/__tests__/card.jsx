import React from 'react';
import renderer from 'react-test-renderer';

import Card from '../card/card';

describe('Card', () => {
  it('renders default type correctly', () => {
    const tree = renderer
      .create(<Card>Unit test</Card>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  ['primary', 'secondary', 'tertiary'].forEach((type) => {
    it(`renders ${type} type correctly`, () => {
      const tree = renderer
        .create(<Card type={type}>Unit test</Card>)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('renders empty card correctly', () => {
    const tree = renderer
      .create(<Card />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
