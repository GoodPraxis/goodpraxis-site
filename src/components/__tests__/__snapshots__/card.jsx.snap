// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders default type correctly 1`] = `
<div
  className="card-component -primary grid-full-sm grid-two-lg"
>
  <div
    className="content"
  >
    Unit test
  </div>
</div>
`;

exports[`Card renders empty card correctly 1`] = `
<div
  className="card-component -primary grid-full-sm grid-two-lg"
>
  <div
    className="content"
  />
</div>
`;

exports[`Card renders primary type correctly 1`] = `
<div
  className="card-component -primary grid-full-sm grid-two-lg"
>
  <div
    className="content"
  >
    Unit test
  </div>
</div>
`;

exports[`Card renders secondary type correctly 1`] = `
<div
  className="card-component -secondary grid-full-sm grid-two-lg"
>
  <div
    className="content"
  >
    Unit test
  </div>
</div>
`;

exports[`Card renders tertiary type correctly 1`] = `
<div
  className="card-component -tertiary grid-full-sm grid-two-lg"
>
  <div
    className="content"
  >
    Unit test
  </div>
</div>
`;
